File: Android/ApiTestUtil/app/src/main/java/cn/com/hvit/apitestutil/Interface/Bean/ApiInterface.java
package cn.com.hvit.apitestutil.Interface.Bean;

import android.util.Log;

import java.util.HashMap;

/**
 * Created by JasonLee on 16/01/25.
 */
public class ApiInterface {
    public String api_name;
    public String api_url;
    public HashMap<String, String> request_params;
    public HashMap<String, String> response_params;
    public String projectName;

    public void init(Object data) {
        HashMap<String, Object> api = (HashMap<String, Object>) data;
        if (api != null) {
            this.api_name = api.get("api_name").toString();
            String unsolvedString = api.get("api_url").toString();

            String[] splited = unsolvedString.split("@");
            Log.i("旧字符串", unsolvedString);
//            for (String splitedString :
//                    splited) {
//                this.api_url += splitedString;
//            }
            this.api_url = splited[0] + "&" + splited[1] + "&" + splited[2];
            Log.i("新字符串", this.api_url);

            this.request_params = (HashMap<String, String>) api.get("request_params");
            this.response_params = (HashMap<String, String>) api.get("response_params");
        }
    }

    public Object toObject() {
        HashMap<String, Object> object = new HashMap<String, Object>();

        object.put("api_name", api_name);
        object.put("api_url", api_url);
        object.put("request_params", request_params);
        object.put("response_params", response_params);

        return object;
    }

    public void logString() {
        Log.i("api_name", this.api_name);
        Log.i("api_url", this.api_url);
        Log.i("request_params", this.request_params.toString());
        Log.i("response_params", this.response_params.toString());
    }
}

File: iOS/ApiTestingUtil/ApiTestingUtil/Interface/Model/ApiInterface.swift
//
//  ApiInterface.swift
//  ApiTestingUtil
//
//  Created by Jason Lee on 16/01/25.
//  Copyright © 2016年 CZLee. All rights reserved.
//

import UIKit

struct ApiInterface {
    private var apiName: String?
    private var apiURL: String?
    var requestParams: [String : AnyObject]?
    var responseParams: [String : AnyObject]?
    var belongToProject: String?
    
    var name: String {
        get {
            return self.apiName ?? ""
        }
    }
    var api_url: String {
        get {
            return self.apiURL ?? ""
        }
    }
    var request_params: [String : AnyObject] {
        get {
            return self.requestParams ?? [String : AnyObject]()
        }
    }
    var response_params: [String : AnyObject] {
        get {
            return self.responseParams ?? [String : AnyObject]()
        }
    }
    
    init(data: [String: AnyObject]) {
        self.apiName = data["api_name"] as? String
        self.apiURL = data["api_url"] as? String
        self.requestParams = data["request_params"] as? [String : AnyObject]
        self.responseParams = data["response_params"] as? [String : AnyObject]
    }
    
    func toDictionary() -> [String : AnyObject] {
        return [
            "api_name" : self.name,
            "api_url" : self.api_url,
            "request_params" : self.request_params,
            "response_params" : self.response_params
        ]
    }
    
    func toString() -> String {
        return ("    {\n    api_name : \(self.name)\n    api_url : \(self.api_url)\n    request_params : \(self.request_params.toString())\n    response_params : \(self.response_params.toString())\n    }\n")
    }
}
